plugins/modules/foreman_compute_resource.py

```diff
@@ -112,133 +150,91 @@
 
 RETURN = ''' # '''
 
-try:
-    from ansible.module_utils.ansible_nailgun_cement import (
-        find_compute_resource,
-        find_organizations,
-        find_locations,
-        ForemanEntityAnsibleModule,
-        naildown_entity_state,
-        sanitize_entity_dict,
-    )
-
-    from nailgun.entities import (
-        AbstractComputeResource,
-        LibvirtComputeResource,
-        OVirtComputeResource,
-        VMWareComputeResource,
-    )
-
-    has_import_error = False
-except ImportError as e:
-    has_import_error = True
-    import_error_msg = str(e)
 
-
-# This is the only true source for names (and conversions thereof)
-name_map = {
-    'name': 'name',
-    'description': 'description',
-    'provider': 'provider',
-    'organizations': 'organization',
-    'locations': 'location',
-    'datacenter': 'datacenter',
-}
+from ansible.module_utils.foreman_helper import ForemanEntityApypieAnsibleModule
 
 
 def get_provider_infos(provider):
     provider_name = provider.lower()
 
     if provider_name == 'libvirt':
-        return {
-            'params': ['url', 'display_type'],
-            'class': LibvirtComputeResource
-        }
+        return {'params': ['url', 'display_type']}
```

Comment:
given `params` is the only thing we return now, maybe rename the function to `get_provider_params` and return the list directly, instead of wrapping it in a dict?

Reply:
Sounds like a good idea.